Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -97,21 +102,44 @@ void OverlayPanel::set_positioning(Positioning positioning) {
   m_positioning = positioning;
 }
 
+bool OverlayPanel::event(QEvent* event) {
+  if(m_is_closed_on_blur) {
+    if(event->type() == QEvent::WindowDeactivate) {
+      if(!is_ancestor(qApp->activeWindow())) {
+        close();
+      }
+    } else if(event->type() == QEvent::WindowActivate) {
+      m_was_activated = true;
+    }
+  } else if(event->type() == QEvent::ParentAboutToChange) {
```

Comment:
Have you tested this branch at any point?

Reply:
No. I'll add a feature into the UiViewer to test re-parenting of the OverlayPanel.